### PR TITLE
impl fn to detemine initial lpt supply amount

### PR DESCRIPTION
**Pull request by taiki1frsh**
Addressed issues: #364 and #370 
beside, impled very simple test for newly added fn with the modification of the test suite setup.
tested in cli as well.

### CODE REVIEW

File: x/derivatives/keeper/lpt_test.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	"github.com/UnUniFi/chain/x/derivatives/types"
+)
+
+// TODO: impl more various situations for the test cases
+func (suite *KeeperTestSuite) TestInitialLiquidityProviderTokenSupply() {
+	assetPrice, err := suite.app.DerivativesKeeper.GetAssetPrice(suite.ctx, TestBaseTokenDenom)
+	if err != nil {
+		suite.Fail("failed to get asset price")
+	}
+	assetMarketCap := assetPrice.Price.Mul(sdk.NewDecFromInt(sdk.OneInt()))
```
Comment (mkXultra) [inline x/derivatives/keeper/lpt_test.go:15]: do not use other func.
Please hard code the data.

File: x/derivatives/keeper/lpt.go [hunk 2]
```diff
@@ -43,31 +44,29 @@ func (k Keeper) GetLPTokenPrice(ctx sdk.Context) sdk.Dec {
 	return k.GetPoolMarketCap(ctx).CalculateLPTokenPrice(k.GetLPTokenSupply(ctx))
 }
 
-func (k Keeper) GetLPTokenAmount(ctx sdk.Context, amount sdk.Coin) (sdk.Coin, sdk.Coin, error) {
+func (k Keeper) DetermineMintingLPTokenAmount(ctx sdk.Context, amount sdk.Coin) (sdk.Coin, sdk.Coin, error) {
```
Comment (mkXultra) [inline x/derivatives/keeper/lpt.go:47]: plz make test

**Review by mkXultra — CHANGES_REQUESTED** [review]

File: x/derivatives/keeper/lpt_test.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	mockPrice := sdk.OneDec()
+	mockAssetPrice := &pftypes.CurrentPrice{
+		MarketId: "uatom:usd",
+		Price:    mockPrice,
+	}
+
+	mockAssetMarketCap := mockPrice.Mul(sdk.NewDecFromInt(sdk.OneInt()))
+
+	initialLPTSupply, fee, err := suite.app.DerivativesKeeper.InitialLiquidityProviderTokenSupply(suite.ctx, mockAssetPrice, mockAssetMarketCap, TestBaseTokenDenom)
+	suite.Require().Condition(initialLPTSupply.Amount.IsPositive)
```
Comment (mkXultra) [inline x/derivatives/keeper/lpt_test.go:21]: check initialLPTSupply value

Comment (taiki1frsh) [reply x/derivatives/keeper/lpt_test.go]: fixed.

**Review by mkXultra — APPROVED** [review]

**Pull request merged**